File: eCinema/eCinema.Services/Database/Entities/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eCinema.Services.Database.Entities
{
    public class Movie
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Title { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        [Range(1, 500)]
        public int DurationMinutes { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Director { get; set; } = string.Empty;
        
        [Required]
        public DateTime ReleaseDate { get; set; }
        
        [Required]
        public int ReleaseYear { get; set; }
        
        public byte[]? Image { get; set; }
        
        [MaxLength(255)]
        public string? TrailerUrl { get; set; }
        
        public float Grade { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public bool IsDeleted { get; set; } = false;
        
        public virtual ICollection<Screening> Screenings { get; set; } = new List<Screening>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
        public virtual ICollection<MovieGenre> Genres { get; set; } = new List<MovieGenre>();
        public virtual ICollection<MovieActor> Actors { get; set; } = new List<MovieActor>();
    }
} 

File: UI/ecinema_desktop/lib/models/movie.dart
import 'package:json_annotation/json_annotation.dart';
import 'actor.dart';
import 'genre.dart';

part 'movie.g.dart';

@JsonSerializable()
class Movie {
  int? id;
  String? title;
  String? description;
  int? durationMinutes;
  String? director;
  DateTime? releaseDate;
  int? releaseYear;
  List<Genre>? genres;
  String? image;
  String? trailerUrl;
  double? grade;
  List<Actor>? actors;
  final bool isDeleted;
  final bool? isComingSoon;

  Movie({
    this.id,
    this.title,
    this.description,
    this.durationMinutes,
    this.director,
    this.releaseDate,
    this.releaseYear,
    this.genres,
    this.image,
    this.trailerUrl,
    this.grade,
    this.actors,
    required this.isDeleted,
    this.isComingSoon,
  });

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);
  Map<String, dynamic> toJson() => _$MovieToJson(this);
}